configs/module.go

```diff
@@ -197,9 +197,25 @@ func (m *Module) appendFile(file *File) hcl.Diagnostics {
 		} else {
 			fqn = addrs.NewLegacyProvider(reqd.Name)
 		}
+		if !fqn.IsDefault() && !fqn.IsLegacy() {
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Invalid provider source",
+				Detail:   fmt.Sprintf("%s is an invalid source. Provider source may only be used with Hashicorp providers.", fqn.String()),
```

Comment:
Hello here is my extremely useful suggestion

```suggestion
				Detail:   fmt.Sprintf("%s is an invalid source. Provider source may only be used with HashiCorp providers.", fqn.String()),
```